examples/aggregator/module/rust/src/data.rs

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::proto::aggregator::SerializedSparseVector;
+use aggregator_common::Monoid;
+use std::collections::HashMap;
+use std::convert::{From, TryFrom};
+
+// Sparce Vectors are stored as Hash Maps.
+#[derive(Clone, Debug)]
+pub struct SparseVector {
+    map: HashMap<u32, f32>,
+}
+
+// Deserializes a Protobuf message into an internal Sparce Vector implementation.
+// If a Protobuf message has duplicated indices, returns an `Error`.
+impl TryFrom<SerializedSparseVector> for SparseVector {
```

Comment:
If I implement a `TryFrom` trait - I cannot use reference, since the trait defines a `fn try_from(value: T) -> Result<Self, Self::Error>` without a reference
Or I just should create a new trait?

Reply:
Does `impl TryFrom<&SerializedSparseVector>` not work?